pkg/controller/clusterbus/controller.go

```diff
@@ -303,9 +305,11 @@ func (c *Controller) syncHandler(key string) error {
 
 	// Finally, we update the status block of the ClusterBus resource to reflect the
 	// current state of the world
-	err = c.updateClusterBusStatus(clusterBus, dispatcherService, dispatcherDeployment, provisionerDeployment)
-	if err != nil {
-		return err
+	if !equality.Semantic.DeepEqual(origClusterBus.Status, clusterBus.Status) {
```

Comment:
This check should occur inside `updateClusterBusStatus` which already make a DeepCopy of the resource before it is updated. The Status object should be updated inside that method and not before.